app/models/round.rb

```diff
@@ -42,4 +48,70 @@ def update_positions
   def set_default_scores
     self.scores ||= {}
   end
+
+  def validate_scores
+    case round_type
+    when 'king'
+      validate_round_type_score(NEGATIVE_VALUE, (0..NEGATIVE_VALUE).step(-400).to_a, -400)
+    when 'ten'
+      validate_round_type_score(POSITIVE_VALUE, (0..POSITIVE_VALUE).step(400).to_a, 400)
+    when 'queens'
+      validate_round_type_score(NEGATIVE_VALUE, (0..NEGATIVE_VALUE).step(-100).to_a, 100)
+    when 'diamonds'
+      expected_value = NEGATIVE_DIAMONDS * game.game_players.count * 2
+      validate_round_type_score(expected_value, (0..NEGATIVE_VALUE).step(-50).to_a, -50)
+    when 'totale_minus', 'totale_plus'
+      validate_totale_scores(round_type)
+    when 'rentz_minus', 'rentz_plus'
+      validate_rentz_scores(round_type)
+    end
```

Comment:
This looks too complicated, and too hard to follow. Lots of un-named parameters, the constants look a bit weird/unclear name

Reply:
This approach is cleaner and/or easier to understand?